partiql-parser/src/scanner.rs

```diff
@@ -117,8 +135,19 @@ impl<'val> PartiQLScanner<'val> {
         self.remainder = self.remainder.consume(start_off + text.len(), pair.end()?);
 
         let content = match pair.as_rule() {
-            Rule::Keyword => Content::Keyword(text),
-            _ => return syntax_error(format!("Unexpected rule: {:?}", pair), pair.start()?.into()),
+            Rule::Keyword => Content::Keyword(text.to_uppercase().into()),
```

Comment:
Another `Cow<str>` opportunity: [`to_uppercase`](https://doc.rust-lang.org/std/primitive.str.html#method.to_uppercase) will also allocate even if the input is already all caps.